cli/cmd/install.go

```diff
@@ -206,60 +212,85 @@ func newCmdInstall() *cobra.Command {
 		"Replicas of the controller to deploy",
 	)
-	cmd.PersistentFlags().StringVar(
+	flags.StringVar(
 		&options.controllerLogLevel, "controller-log-level", options.controllerLogLevel,
 		"Log level for the controller and web components",
 	)
-	cmd.PersistentFlags().BoolVar(
+	flags.BoolVar(
 		&options.proxyAutoInject, "proxy-auto-inject", options.proxyAutoInject,
 		"Enable proxy sidecar auto-injection via a webhook (default false)",
 	)
-	cmd.PersistentFlags().BoolVar(
+	flags.BoolVar(
 		&options.highAvailability, "ha", options.highAvailability,
 		"Experimental: Enable HA deployment config for the control plane (default false)",
 	)
-	cmd.PersistentFlags().Int64Var(
+	flags.Int64Var(
 		&options.controllerUID, "controller-uid", options.controllerUID,
 		"Run the control plane components under this user ID",
 	)
-	cmd.PersistentFlags().BoolVar(
+	flags.BoolVar(
 		&options.disableH2Upgrade, "disable-h2-upgrade", options.disableH2Upgrade,
 		"Prevents the controller from instructing proxies to perform transparent HTTP/2 upgrading (default false)",
 	)
-	cmd.PersistentFlags().StringVar(
+	flags.StringVar(
 		&options.identityOptions.trustDomain, "identity-trust-domain", options.identityOptions.trustDomain,
 		"Configures the name suffix used for identities.",
 	)
-	cmd.PersistentFlags().StringVar(
+	flags.StringVar(
 		&options.identityOptions.trustPEMFile, "identity-trust-anchors-file", options.identityOptions.trustPEMFile,
 		"A path to a PEM-encoded file containing Linkerd Identity trust anchors (generated by default)",
 	)
-	cmd.PersistentFlags().StringVar(
+	flags.StringVar(
 		&options.identityOptions.crtPEMFile, "identity-issuer-certificate-file", options.identityOptions.crtPEMFile,
 		"A path to a PEM-encoded file containing the Linkerd Identity issuer certificate (generated by default)",
 	)
-	cmd.PersistentFlags().StringVar(
+	flags.StringVar(
 		&options.identityOptions.keyPEMFile, "identity-issuer-key-file", options.identityOptions.keyPEMFile,
 		"A path to a PEM-encoded file containing the Linkerd Identity issuer private key (generated by default)",
 	)
-	cmd.PersistentFlags().DurationVar(
+	flags.DurationVar(
 		&options.identityOptions.clockSkewAllowance, "identity-clock-skew-allowance", options.identityOptions.clockSkewAllowance,
 		"The amount of time to allow for clock skew within a Linkerd cluster",
 	)
-	cmd.PersistentFlags().DurationVar(
+	flags.DurationVar(
 		&options.identityOptions.issuanceLifetime, "identity-issuance-lifetime", options.identityOptions.issuanceLifetime,
 		"The amount of time for which the Identity issuer should certify identity",
 	)
-	return cmd
+
+	return flags
+}
+
+func (options *installOptions) recordFlags(flags *pflag.FlagSet) {
+	if flags == nil {
+		return
+	}
+
+	flags.VisitAll(func(f *pflag.Flag) {
+		if f.Changed {
+			switch f.Name {
+			case "ignore-cluster", "linkerd-version":
+				// Thse flags don't make sense to record.
```

Comment:
typo: These

Also how about including here as well `api-addr`, `context`, `kubeconfig` and `verbose`? Most likely these are particular to some user, so shouldn't be stored in the config shared by all.

Reply:
We only have to handle flags set by options.flagSet or their children now! This was the main rational in building FlagSet instead of appending to PersistenFlags.

I've confirmed this with manual testing.

Reply:
And what do you think about `identity-issuer-certificate-file`,  `identity-issuer-key-file` and `identity-trust-anchors-file`? I believe `linkerd upgrade` will ignore them but they don't belong in the shared config IMO.

Reply:
Good call. These should be omitted as well.

Reply:
PTAL. I moved these issuer flags onto a separate flagset so they are just never considered to be recorded (and also means we won't expose them from upgrade).